datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java

```diff
@@ -31,6 +31,8 @@
   // timestamp of when the record was sent to transport provider
   private Optional<Long> _eventsSendTimestamp = Optional.empty();
 
+  private Optional<Long> _eventsAuditTimestamp = Optional.empty();
```

Comment:
Can you add a comment explaining this field comes from the Kafka header? Also my mistake, we probably shouldn't have the word "audit" in the name. How about renaming it to: `_eventsKafkaHeaderTimestamp`?

Reply:
I was kind of reluctant to add the word audit here too. I guess you're right. Will try to rephrase everything in more generic terms.